src/session/sessionReducer.js

```diff
@@ -56,6 +57,18 @@ export type SessionState = {|
   pushToken: string | null,
 
   debug: Debug,
+
+  /**
+   * Whether `ServerCompatNotice` (which we'll add soon) has been
+   *   dismissed this session.
+   *
+   * We put this in `SessionState` deliberately, so that users see the
+   * notice on every startup until the server is upgraded. That's a
```

Comment:
What's the mechanism by which this gets reset?

Reply:
Good point; I'll mention it in the comment. `state.session` isn't persisted to the disk, and I've set `state.session.hasDismissedServerCompatNotice` to be `false` by default. That means that on startup it'll be initialized to `false`, no matter what its value was in the previous session.